apps/language_server/lib/language_server/server.ex

```diff
@@ -805,13 +805,67 @@ defmodule ElixirLS.LanguageServer.Server do
   end
 
   defp get_test_code_lenses(state, uri, source_file) do
-    if state.settings["enableTestLenses"] == true do
+    get_test_code_lenses(
+      state,
+      uri,
+      source_file,
+      state.settings["enableTestLenses"],
+      Mix.Project.umbrella?()
+    )
+  end
+
+  defp get_test_code_lenses(_state, _uri, _source_file, false, _), do: {:ok, []}
+
+  defp get_test_code_lenses(state, uri, source_file, true = _enabled, true = _umbrella) do
+    file_path = SourceFile.path_from_uri(uri)
+
+    Mix.Project.apps_paths()
+    |> Enum.find(fn {_app, app_path} -> String.contains?(file_path, app_path) end)
+    |> case do
+      nil ->
+        {:ok, []}
+
+      {app, app_path} ->
+        if is_test_file?(file_path, state, app, app_path) do
+          CodeLens.test_code_lens(uri, source_file.text, "#{state.project_dir}/#{app_path}")
+        else
+          {:ok, []}
+        end
+    end
+  end
+
+  defp get_test_code_lenses(state, uri, source_file, true = _enabled, false = _umbrella) do
+    file_path = SourceFile.path_from_uri(uri)
```

Comment:
this call will fail if URI is not in `file:` schema (e.g. `git:`, `untitled:`). You need to rescue and return `[]`